Code/Tools/AssetProcessor/native/utilities/SimpleStatsCapture.cpp

```diff
@@ -0,0 +1,392 @@
+                    AZStd::string_view jobKey = tokens[2];
+                    AZStd::string_view platformName = tokens[3];
+
+                    // synthesize a stat to record process time accumulated by job key platform
+                    {
+                        AZStd::string newStatKey = AZStd::string::format("ProcessJobsByPlatform,%.*s", AZ_STRING_ARG(platformName));
+                        auto insertion = m_stats.insert(newStatKey);
+                        StatsEntry& statToSynth = insertion.first->second;
+                        statToSynth.m_cumulativeTime += statistic.m_cumulativeTime;
+                        statToSynth.m_operationCount += statistic.m_operationCount;
+                        if (insertion.second)
+                        {
+                            allProcessJobsByPlatform.push_back(newStatKey);
+                        }
+                    }
+
+                    // synthesize a stat to record process time accumulated job key total across all platforms
+                    {
+                        AZStd::string newStatKey = AZStd::string::format("ProcessJobsByJobKey,%.*s", AZ_STRING_ARG(jobKey));
+                        auto insertion = m_stats.insert(newStatKey);
+                        StatsEntry& statToSynth = insertion.first->second;
+                        statToSynth.m_cumulativeTime += statistic.m_cumulativeTime;
+                        statToSynth.m_operationCount += statistic.m_operationCount;
+                        if (insertion.second)
+                        {
+                            allProcessJobsByJobKey.push_back(newStatKey);
+                        }
+                    }
+                    // synthesize a stat to track total processjob times:
+                    {
+                        StatsEntry& statToSynth = m_stats["ProcessJobsTotal"];
+                        statToSynth.m_cumulativeTime += statistic.m_cumulativeTime;
+                        statToSynth.m_operationCount += statistic.m_operationCount;
+                    }
+                }
+                else if (AZ::StringFunc::StartsWith(statKey, "HashFile,", true))
+                {
+                    allHashFiles.push_back(statKey);
+                    // processjob has the format ProcessJob,sourcename,jobkey,platformname
+                    // synthesize a stat to track total hash times:
+                    StatsEntry& statToSynth = m_stats["HashFileTotal"];
+                    statToSynth.m_cumulativeTime += statistic.m_cumulativeTime;
+                    statToSynth.m_operationCount += statistic.m_operationCount;
+                }
+            }
+            
+            auto sortByTimeDescending = [&](const AZStd::string& s1, const AZStd::string& s2)
+            {
+                return this->m_stats[s1].m_cumulativeTime > this->m_stats[s2].m_cumulativeTime;
+            };
+            
+            auto printStat = [&](const char* name, duration milliseconds, int64_t count)
```

Comment:
It would help keep the length of individual functions down if these lambas are moved to be member functions.

Reply:
Sure, I'll move these to be member functions.